presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/IntegerIdentityPartitioner.java

```diff
@@ -37,7 +37,9 @@ public int numPartitions()
     @Override
     public int getPartition(Object key)
     {
-        int partition = requireNonNull((Integer) key, "key is null");
+        requireNonNull(key, "key is null");
+        MutablePartitionId mutablePartitionId = (MutablePartitionId) key;
```

Comment:
nit: i think this class should be renamed to be something like `PrestoSparkPartitioner` now 😃